### ISSUE TITLE
NNcase producing padding irrespective of valid or same for tensorflowlite model from Keras

### ISSUE THREAD

**Issue by krishnak** [kendryte/nncase#89]
I am trying to create a model using keras 2 which uses tensor flow as back end. 

Its a very simple code to test nncase conversion

```
# Initializing the CNN
classifier = Sequential()

#lossbranch_1 = Sequential()
# First convolution layer C1
classifier.add(Convolution2D(16, (3, 3),strides=(2,2),padding='valid',dilation_rate=(1, 1),use_bias=True, input_shape=(240, 320, 3)))
classifier.add(Activation('relu'))
# Second convolution layer C2
c2 = Convolution2D(16, (3, 3),strides=(1,1),padding='same',use_bias=True, activation='relu')
classifier.add(c2)
```
Attached is the keras model converted to tensorflow lite using

`tflite_convert --output_file kkfirstmodel.tflite --keras_model_file firstmodel.h5 `

The output graph of kmodel is having a padding operation before the first convolution. I want to avoid it, this padding operation appears irrespective of whether I use padding="valid" or padding ="same" 

Can you advise what needs to be done to avoid it? 

The caffe model of ultra fast detector has a similar first convolution layer, the kmodel graph for it doesn't produce the padding operation for it.

[firstmodel5.pdf](https://github.com/kendryte/nncase/files/4172013/firstmodel5.pdf)

[kkfirstmodel.zip](https://github.com/kendryte/nncase/files/4172037/kkfirstmodel.zip)

**Comment by sunnycase** [comment]
KPU only support conv2d with pad [[0,0],[0,0],[1,1],[1,1]] (NCHW), so nncase will automatically add necessary pad and crop before and after `kpu_conv2d`.
If you want to avoid the pad, you can add a pad [[0,0],[0,0],[1,1],[1,1]] before your conv2d and use valid padding mode.

**Comment by krishnak** [comment]
Thanks for the clarification. The following code solves the problem in keras
```
classifier = Sequential()
classifier.add(ZeroPadding2D(1, input_shape=(240, 320, 3)))
# First convolution layer C1
classifier.add(Convolution2D(16, (3, 3),strides=(2,2),padding='valid',use_bias=True,activation='relu'))
```
